src/python/pants/option/global_options.py

```diff
@@ -1662,6 +1662,64 @@ class GlobalOptions(BootstrapOptions, Subsystem):
         ),
     )
 
+    _use_deprecated_pex_binary_run_semantics = BoolOption(
+        "--use-deprecated-pex-binary-run-semantics",
+        default=True,
+        help=softwrap(
+            """
+            If `true`, `run`ning a `pex_binary` will run your firstparty code by copying sources to
+            a sandbox (while still using a PEX for thirdparty dependencies). Additionally, you can
+            refer to the `pex_binary` using the value of its `entry_point` field (if it is a filename).
+
+            If `false`, `run`ning a `pex_binary` will build the PEX via `package` and run it directly.
+            This makes `run` equivalent to using `package` and running the artifact. Additionally,
+            the binary must be `run` using the `pex_binary`'s address, as passing a filename to `run`
+            will run the `python_source`.
+
+            Note that support has been added to Pants to allow you to `run` any `python_source`,
+            so setting this to `true` should be reserved for maintaining backwards-compatibility
+            with previous versions of Pants. Additionally, you can remove any `pex_binary` targets
+            that exist solely for running Python code (and aren't meant to be packaged).
+            """
+        ),
+        removal_version="2.15.0.dev0",
```

Comment:
If you want this, set `removal_start_version="2.14.0.dev0"`. Otherwise we will complain when you explicitly set the value.

Reply:
`TypeError: __new__() got an unexpected keyword argument 'removal_start_version'`

Reply:
Hm, then should remove this line